### ISSUE TITLE
NullReferenceException while navigating to a UserControl has poor exception message

### ISSUE THREAD

**Issue by TroelsL** [microsoft/microsoft-ui-xaml#5075]
**Describe the bug**
When calling the method Navigate (or any of its overloads) on a Microsoft.UI.Xaml.Controls.Frame with a parameter that is not a type that inherits from Page, a NullReferenceException is thrown. The exception message is very generic and could be improved.

**Steps to reproduce the bug**

1. Add a Frame (MyFrame) anywhere in an application
2. Make the following method call: MyFrame.Navigate(typeof(UserControl))

**Expected behavior**

An ArgumentException with a descriptive text would be better if the non-generic solution is needed. Changing the signature of the method to the suggestion below would be better still in some scenarios: 

bool Navigate<T>(object parameter = null) where T : Page

**Screenshots**

**Version Info**

NuGet package version: 
[Microsoft.WinUI 3.0.0-preview4.210210.4] 
Also happened in UWP according to this SO post: https://stackoverflow.com/questions/38620255/uwp-frame-null-reference

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| No | Yes |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| May 2020 Update (19041)            | Yes   |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |

**Additional context**

**Comment by Jaiganeshkumaran** [comment]
Frame is designed to navigate to Page instances, not UserControls.

**Comment by TroelsL** [comment]
@Jaiganeshkumaran I am aware. This is a suggestion to improve the exception, not change the behaviour.

**Comment by asklar** [comment]
re: `bool Navigate(T parameter = null) where T : Page`, WinRT doesn't support restrictions on generics so this could only be implemented as a special case in cswinrt, and _not_ at the winui API layer. I don't see the stack trace/exception msg in your post but it sounds like it could be made better.

**Comment by chrisglein** [comment]
Would be nice if we were going back to change design of the signature here, but absent of this we can't fix it. Closing out the issue. Appreciate the suggestion though.

**Comment by TroelsL** [comment]
Fair enough regarding the signature, but an ArgumentException("Type must be a subclass of Page") would still be an improvement over a NullReferenceException without any message.

**Comment by chrisglein** [comment]
> Fair enough regarding the signature, but an ArgumentException("Type must be a subclass of Page") would still be an improvement over a NullReferenceException without any message.

Fair suggestion. The internal mechanism here must be passing the failed cast on as a null value resulting in the less-than-helpful suggestion. Fix would be to offer up a better exception type. Reactivated.

**Comment by AtomicBlom** [comment]
I just spent a lot longer than I wanted to trying to diagnose this issue as well, a more explicit exception message would be much appreciated. Thank you TroelsL for including the solution.

**Comment by DamianMehers** [comment]
FWIW I've just hit this issue navigating to a Window rather than a Page by mistake.  A better error message would definitely have helped.

**Comment by DamianMehers** [comment]
Please don't close without resolving.

**Comment by MPITech** [comment]
I just hit this problem as well.  Somewhere in the documentation it said I could navigate to a UserControl, I forget where, and no matter what I tried I kept getting a NullReferenceException and couldn't track it down until I found this page.  There should be a more descriptive error message.
